### PR TITLE
initial PR, added github login, refactoring of the style, code; added MongoDb for tasks

### PR DESCRIPTION
**Pull request by sashameniakin**

### CODE REVIEW

File: components/ArchiveCard.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <StyledP>{title}</StyledP>
+        <StyledP id>{link}</StyledP>
+        <StyledP>{blockchain}</StyledP>
+        <StyledP>{deadline}</StyledP>
+        <StyledP>{description}</StyledP>
+        <StyledButton onClick={() => handleDelete(id)}>
+          <StyledImage alt="delete" src={Delete} width="25px" height="25px" />
+        </StyledButton>
+      </StyledArchiveCard>
+    </>
```
Comment (stefankorder) [inline components/ArchiveCard.js:30]: You don't need a Fragment for this case:

```suggestion
      <StyledArchiveCard>
        <StyledP id>{id}</StyledP>
        <StyledP>{title}</StyledP>
        <StyledP id>{link}</StyledP>
        <StyledP>{blockchain}</StyledP>
        <StyledP>{deadline}</StyledP>
        <StyledP>{description}</StyledP>
        <StyledButton onClick={() => handleDelete(id)}>
          <StyledImage alt="delete" src={Delete} width="25px" height="25px" />
        </StyledButton>
      </StyledArchiveCard>
```

File: components/CoinCard.js [hunk 2]
```diff
@@ -22,24 +23,18 @@ export default function CoinCard({
...
-      <StyledDiv> {rank}</StyledDiv>
-      <StyledDiv> {name}</StyledDiv>
-      <StyledDiv> {price.toFixed(2)}</StyledDiv>
-      <StyledDiv> {market_cap.toFixed(2)}</StyledDiv>
-      <StyledDiv> {volume.toFixed(2)}</StyledDiv>
+      <div> {rank}</div>
+      <div> {name}</div>
+      <div> {price.toFixed(2)}</div>
+      <div> {(market_cap.toFixed(2) / 1000000000).toFixed(2)}</div>
+      <div> {(volume.toFixed(2) / 1000000).toFixed(2)}</div>
```
Comment (stefankorder) [inline components/CoinCard.js:30]: Please use don't use `<div>` tags, use other more symantic tags like `<p>`.

File: components/Diagram.js [hunk 3]
```diff
@@ -84,7 +85,7 @@ const Diagram = ({diagram, coins}) => {
       ],
       labels: labels,
     });
-  }, [diagram]);
+  }, [diagram, coins]);
 
   /*   const textCenter = {
```
Comment (stefankorder) [inline components/Diagram.js:90]: Please delete code you don't use anymore.

File: components/Navbar.js [hunk 4]
```diff
@@ -206,10 +217,7 @@ const StyledDiv = styled.div`
...
   height: 35px;
   background-color: ${props =>
-    props.active === true ? "rgba(255, 123, 137)" : ""};
-`;
-
-const StyledButton = styled.button`
-  background-color: transparent;
-  border: none;
+    props.active === true
+      ? /* "rgba(165, 202, 210, 0.5)" */ "rgba(255, 123, 137, 0.3)"
```
Comment (stefankorder) [inline components/Navbar.js:221]: Please delete code you don't use anymore.

File: components/Popup_bookmarked.js [hunk 5]
```diff
@@ -13,29 +17,34 @@ function Popup(props) {
...
-          </StyledHead>
+          </StyledCloseButton>
+          <StyledH2>YOUR WATCHLIST</StyledH2>
+          <StyledCard header>
+            <div />
+            <div>#</div>
+            <div>NAME</div>
+            <div>PRICE($)</div>
+            <div>MARKET CAP($bln.)</div>
+            <div>VOLUME (24h)($mln.)</div>
```
Comment (stefankorder) [inline components/Popup_bookmarked.js:30]: Please use don't use `<div>` tags, use other more symantic tags like `<p>`.

File: components/styled/FeatureBackground.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+import styled from "styled-components";
+
+const FeatureBackground = styled.div`
+  width: ${props => (props.welcome || props.login ? "100%" : "98%")};
+  height: ${props =>
+    props.welcome ||
+    /* props.transactions || */
```
Comment (stefankorder) [inline components/styled/FeatureBackground.js:7]: Please delete code you don't use anymore.

File: context/context.js [hunk 7]
```diff
@@ -333,6 +355,108 @@ export function FundsProvider({children}) {
...
+    const result = await response.json();
+    setShouldReload(true);
+  };
+
+  const handleDelete = async id => {
+    try {
+      const response = await fetch(`/api/tasks/${id}`, {
+        method: "DELETE",
+      });
+      console.log(id);
```
Comment (stefankorder) [inline context/context.js:414]: Please delete this `console.log`. In your final PR there should be no console logs you are just using for development.

File: pages/funds/index.js [hunk 8]
```diff
@@ -114,47 +118,46 @@ export default function Funds() {
...
-            <div>PRICE</div>
-            <div>DATE</div>
-            <div>TIME</div>
+            <StyledP>ID</StyledP>
+            <StyledP>ACTION</StyledP>
+            <StyledP>COIN</StyledP>
+            <StyledP>AMT.</StyledP>
+            <StyledP>PRICE($)</StyledP>
+            <StyledP>DATE</StyledP>
+            <StyledP>TIME</StyledP>
```
Comment (stefankorder) [inline pages/funds/index.js:160]: 👍🏻

File: pages/home/index.js [hunk 9]
```diff
@@ -40,6 +44,7 @@ export default function Home() {
         console.log(error.message);
       }
     };
+    console.log(coinData);
```
Comment (stefankorder) [inline pages/home/index.js:47]: Take a look at my comment above about console logs.

File: pages/login/index.js [hunk 10]
```diff
@@ -1,29 +1,119 @@
...
+            <Image alt="github_logo" src={github_logo}></Image>
+            <p>Login with </p>
+            <Image alt="github" src={github}></Image>
+          </StyledButtonGitHub>
+        </StyledBackground>
+        {/*    <Link href="/profile">
+          <StyledButtonMain test>Go to app (without login)</StyledButtonMain>
+        </Link> */}
+      </FeatureBackground>
     </>
```
Comment (stefankorder) [inline pages/login/index.js:73]: You don't need a fragment for this case. So delete `<>` and `</>`.

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work so far! Take a look at my comments. Also try to find all console logs you don't really need. Think also about the css units you use. In generell: Maybe you could use more relative units to give your application a more responsive touch.

File: context/context.js (hunk 7 above)
Comment (sashameniakin) [reply context/context.js:414]: done!

File: pages/home/index.js (hunk 9 above)
Comment (sashameniakin) [reply pages/home/index.js:47]: done!

File: pages/login/index.js (hunk 10 above)
Comment (sashameniakin) [reply pages/login/index.js:73]: done!

File: components/styled/FeatureBackground.js (hunk 6 above)
Comment (sashameniakin) [reply components/styled/FeatureBackground.js:7]: done!

File: components/Navbar.js (hunk 4 above)
Comment (sashameniakin) [reply components/Navbar.js:221]: done!

File: components/ArchiveCard.js (hunk 1 above)
Comment (sashameniakin) [reply components/ArchiveCard.js]: done!

File: components/Diagram.js (hunk 3 above)
Comment (sashameniakin) [reply components/Diagram.js]: done!

File: components/CoinCard.js (hunk 2 above)
Comment (sashameniakin) [reply components/CoinCard.js:30]: done!

**Pull request merged**